### PR TITLE
porting pubsub trigger

### PR DESCRIPTION
**Pull request by thatfiredev**

### CODE REVIEW

File: 2nd-gen/pubsub-helloworld/functions/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * limitations under the License.
+ */
+"use strict";
+
+// [START import]
+const {onMessagePublished} = require("firebase-functions/v2/pubsub");
+const logger = require("firebase-functions/logger");
+// [END import]
+
+// [START helloWorld]
```
Comment (egilmorez) [inline 2nd-gen/pubsub-helloworld/functions/index.js:23]: I think we do want to prepend tag names with "v2."

As it is here, the snippet name is created as "js_helloWorld" which is rather general.

Conversely, a snippet name like "js_v2httpReadQueryParam" is more recognizable as a CF3 v2 snippet.

**Review by egilmorez — COMMENTED** [review]
One global comment, which I think Jeff Huleatt will second. 

Thanks for sending!

File: 2nd-gen/pubsub-helloworld/functions/index.js (hunk 1 above)
Comment (thatfiredev) [reply 2nd-gen/pubsub-helloworld/functions/index.js]: Done

File: 2nd-gen/pubsub-helloworld/functions/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+> The first time you call `firebase deploy` on a new project with Functions will take longer than usual.
+
+
+## Try the sample
+
+Once deployed, to try the sample use the `gcloud` CLI to publish a message to the `topic-name` topic:
+
+```
+gcloud alpha pubsub topics publish topic-name --message='YourName'
```
Comment (jhuleatt) [inline 2nd-gen/pubsub-helloworld/functions/README.md:66]:
```suggestion
gcloud pubsub topics publish topic-name --message='YourName'
```

File: 2nd-gen/pubsub-helloworld/functions/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+Open the Functions logs in the Firebase Console, you should see a messages that reads "Hello YourName".
+
+Then you can also publish a message to the `another-topic-name` topic using JSON data:
+
+```
+gcloud alpha pubsub topics publish another-topic-name --message='{"name":"YourName"}'
```
Comment (jhuleatt) [inline 2nd-gen/pubsub-helloworld/functions/README.md:74]:
```suggestion
gcloud pubsub topics publish another-topic-name --message='{"name":"YourName"}'
```

File: 2nd-gen/pubsub-helloworld/functions/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+exports.hellopubsub = onMessagePublished("topic-name", (event) => {
+  // [END v2trigger]
+  // [START v2readBase64]
+  // Decode the PubSub Message body.
+  const message = event.data.message;
+  const messageBody = message.data ?
+        Buffer.from(message.data, "base64").toString() :
+        null;
+    // [END v2readBase64]
+    // Print the message in the logs.
```
Comment (jhuleatt) [inline 2nd-gen/pubsub-helloworld/functions/index.js:38]:
```suggestion
  // [END v2readBase64]
  // Print the message in the logs.
```

**Review by jhuleatt — APPROVED** [review]
LGTM! Just left suggestions for a couple of small fixes

File: 2nd-gen/pubsub-helloworld/functions/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+
+## Contributing
+
+We'd love that you contribute to the project. Before doing so please read our [Contributor guide](../../../CONTRIBUTING.md).
+
+
+## License
+
+© Google, 2016. Licensed under an [Apache-2](../../../LICENSE) license.
```
Comment (thatfiredev) [inline 2nd-gen/pubsub-helloworld/functions/README.md:95]:
```suggestion
© Google, 2022. Licensed under an [Apache-2](../../../LICENSE) license.
```

**Review by egilmorez — APPROVED** [review]
Thanks!

**Pull request merged**